src/widget/linear_layout.rs

```diff
@@ -28,14 +28,21 @@ use super::{contexts::LifeCycleCx, EventCx, LayoutCx, LifeCycle, PaintCx, Pod, U
 /// widget.
 pub struct LinearLayout {
     pub children: Vec<Pod>,
+    pub tree_mutations: Vec<tree_structure::SpliceMutation>,
     pub spacing: f64,
     pub axis: Axis,
 }
```

Comment:
Overall, I'd would really like if `SpliceMutation` and everything related to it was kept out of the `widget` module. It would match how things will look after the Masonry refactor.

This means a few things:

- Nix most of the changes to `src/widget/*.rs`
- In `src/widget/tree_structure.rs`, keep `TreeStructure` and move `SpliceMutation` and `TreeTrackerSplice` out.
- Move the `&mut TreeStructure` out of `CxState` and into `xilem::view::Cx`.
- Basically handle updating the TreeStructure from the view side.

Does that seem feasible to you?

Reply:
Sure I guess that makes sense.
One thing though, I think it makes sense to at least have readable access in `CxState` (e.g. to target the right accessibility node as is done in `Event::TargetedAccessibilityAction`).